### PR TITLE
[SPARK-14996][SQL] Add TPCDS Benchmark Queries for SparkSQL

### PR DESCRIPTION
**Pull request by sameeragarwal**
## What changes were proposed in this pull request?

This PR adds support for easily running and benchmarking a set of common TPCDS queries locally in SparkSQL.
## How was this patch tested?

N/A

### CODE REVIEW

**Comment by sameeragarwal** [comment]
cc @davies

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/TPCDSBenchmark.scala [hunk 1]
```diff
@@ -0,0 +1,1225 @@
...
+    TPCDS Snappy (scale = 5):           Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+    -------------------------------------------------------------------------------------------
+    ss_max                                   2305 / 2731          6.2         160.0       1.0X
+     */
+
+  ).toArray
+
+  val tables = Seq("customer", "customer_address", "customer_demographics", "date_dim",
+    "household_demographics", "inventory", "item", "promotion", "store", "catalog_sales",
+    "web_sales", "store_sales")
```
Comment (davies) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/TPCDSBenchmark.scala]: Why not have all the tables?

Comment (sameeragarwal) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/TPCDSBenchmark.scala]: any particular reason?

**Comment by sameeragarwal** [comment]
test this please

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/TPCDSBenchmark.scala (hunk 1 above)
Comment (rxin) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/TPCDSBenchmark.scala]: It's good to have them for sure in the future. Anyway this is OK for now.

**Comment by rxin** [comment]
Going to merge this for now.

**Pull request closed without merging**